components/tag-list/tag-list-item.js

```diff
@@ -9,12 +10,46 @@ class TagListItem extends TagListItemMixin(LitElement) {
 			 * REQUIRED: Text to display
 			 * @type {string}
 			 */
-			text: { type: String }
+			text: { type: String },
+			/**
+			 * Optional: Text to display in tooltip.
+			 * Tooltip will also include text property value if truncated.
+			 * @type {string}
+			 */
+			description: { type: String },
+			_isTruncated: { attribute: false }
 		};
 	}
 
+	constructor() {
+		super();
+		this._isTruncated = false;
+	}
+
+	disconnectedCallback() {
+		super.disconnectedCallback();
+		if (this._resizeObserver) this._resizeObserver.disconnect();
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (this.description) {
+			// observe resize events so we can update tooltip text when truncated
+			const itemContent = this.shadowRoot.querySelector('.tag-list-item-content');
+			this._onListItemResize();
+			this._resizeObserver = new ResizeObserver(() => this._onListItemResize());
+			this._resizeObserver.observe(itemContent);
```

Comment:
The tooltip [has some pretty complex](https://search.d2l.dev/xref/BrightspaceUI/core/components/tooltip/tooltip.js?r=56fecdc9#909) logic already to determine if the target is truncating... so it's unfortunate that in this case we have to run it twice. Tooltips of course have to deal with any kind of target, whereas this code is very specific to tag list items.

How terrible would it be if we just always showed the text + description in the tooltip? Then all this code could go away and we'd just rely on tooltip to show when things are truncated if there's no description, or always show if there is a description.

Reply:
Ideally we keep them separate when it is not truncated as that follows our current design. This shouldn't really impact performance I would think as it is a simple resize observer. Would you block this change if we kept it this way?

Reply:
Yeah, tooltip also delays the truncation calculations until the moment when we are about to show the tooltip (or not), rather than on every resize.  That logic was originally added for this use case, I wonder if we can instead just expose that calculation instead of doing it twice.

> How terrible would it be if we just always showed the text + description in the tooltip? Then all this code could go away and we'd just rely on tooltip to show when things are truncated if there's no description, or always show if there is a description.

I was wondering that too, it results in a slightly repetitive tooltip but might be more consistent of an experience for the user.  @geurts do you have thoughts here on how general `d2l-tag-list-items` should behave when their tooltip text is customized?

Reply:
Potential solution here: https://github.com/BrightspaceUI/core/pull/2772/commits/11b94cf6aeff2258d34d06c5a35e09e709b53528
Just remove the `?show-truncated-only` tooltip check and use the new truncated check I added in `d2l-list-item` to control whether a tooltip is added

Reply:
I think it does the check regardless of whether the user interacts though ya? I think we would like to defer as much work as possible until needed, especially for components that could be rendered on the page many times. In the future we might even want to avoid rendering such aspects of the component until necessary, but that's a much bigger thing.